kognic-io/kognic/io/tools/scene_input.py

```diff
@@ -0,0 +1,147 @@
+            pre_annotation = scenes_map[input_result.scene_uuid].pre_annotation
+            if pre_annotation:
+                pre_anno_result = _create_pre_annotation(client, input_result.scene_uuid, pre_annotation)
+                input_result.update(pre_anno_result)
+
+        if input_result.status == CreationStatus.CREATED:
+            input_from_scene_result = _create_input_from_scene(resource, input_result.scene_uuid, project, **kwargs)
+            input_result.update(input_from_scene_result)
+        yield input_result
+
+
+def _start_scene_jobs(
+    resource: CreateableIOResource,
+    scenes: List[Scene],
+    dryrun: bool = False,
+    feature_flags: Optional[FeatureFlags] = None
+) -> List[SceneUuid]:
+    # Starts scene jobs. Don't catch validation errors. The returned list must have the same order as the scenes
+    return [resource.create(scene, dryrun=dryrun, feature_flags=feature_flags).input_uuid for scene in scenes]
+
+
+def _wait_for_scene_jobs(
+    client: KognicIOClient,
+    scene_uuids: List[SceneUuid],
+    wait_timeout: float,
+    sleep_time: float
+) -> Generator[CreationResult, None, None]:
+    def poll_input_status(su: str) -> CreationResult:
+        """Wait for scene processing"""
+        start_time = time.time()
+        while time.time() - start_time < wait_timeout:
+            responses = client.input.get_inputs_by_uuids(input_uuids=[su])
+
+            if not responses:
+                return CreationResult(scene_uuid=su, status=CreationStatus.CREATED)
+
+            response = responses[0]
+            if response.status in [InputStatus.Failed]:
+                return CreationResult(scene_uuid=su, status=CreationStatus.FAILED, error=CreationError(stage=CreationStage.SCENE, message=response.error_message))
+
+            # Check timeout
+            time.sleep(sleep_time)
+        return CreationResult(scene_uuid=su, status=CreationStatus.FAILED, error=CreationError(stage=CreationStage.SCENE, message='Scene creation timed out'))
+
+    # Wait for scene jobs to complete. Yield creation results as they are or the process times out
+    # Having some good wait strategy here will be important. Think about big vs small scenes and the number of scenes
+    with ThreadPoolExecutor(max_workers=5) as executor:
+        features = [executor.submit(poll_input_status, su) for su in scene_uuids]
+        for future in as_completed(features):
+            yield future.result()
+
+def _create_pre_annotation(client: KognicIOClient, scene_uuid: str, pre_annotation: OpenLabelAnnotation) -> CreationResult:
+    # Try to create a pre-annotation and return a creation result
+    try:
+        client.pre_annotation.create(scene_uuid, pre_annotation, dryrun=False)
+        return CreationResult(scene_uuid=scene_uuid, status=CreationStatus.CREATED)
+    except HTTPError as error:
+        logger.error(f"Failed to create pre-annotation for scene {scene_uuid}. {error}")
```

Comment:
This will make errors clearer. It's not the full stacktrace but it's a bit more than when parsing to a string
```suggestion
        logger.error(f"Failed to create pre-annotation for scene {scene_uuid}. {error}", exc_info = error)
```